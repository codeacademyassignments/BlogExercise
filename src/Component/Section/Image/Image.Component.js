import React from 'react';
import '../../index.css';


const Image = (props)=>{
  const image = require(`../../../Images/${props.imageUrl}`);
  return (
  <div className='img' style={{backgroundImage:`url(${image})`}}></div>
)};

export default Image;