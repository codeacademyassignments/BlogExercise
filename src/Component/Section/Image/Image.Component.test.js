import React from 'react';
import renderer from 'react-test-renderer';
import Image from './Image.Component';
import {shallow} from 'enzyme';

describe('snapshot testing',()=>{
  it('should match the snapshot',()=>{
  const tree = renderer.create(<Image imageUrl='outdoor.png'/>);
  expect(tree).toMatchSnapshot();
  });
})

describe('style attributes',()=>{
  const wrapper = shallow(<Image imageUrl={'outdoor.png'}/>);
  it('should set the component attribute to correct value',()=>{
    expect(wrapper.find('.img').prop('style')).toHaveProperty('backgroundImage','url(outdoor.png)');
  })
})