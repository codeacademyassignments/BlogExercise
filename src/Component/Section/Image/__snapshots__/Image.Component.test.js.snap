// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing should match the snapshot 1`] = `
<div
  className="img"
  style={
    Object {
      "backgroundImage": "url(outdoor.png)",
    }
  }
/>
`;
